### PR TITLE
bug: now longer allow migrate the object store twice

### PR DESCRIPTION
**Pull request by camilamacedo86**
#### What this PR does / why we need it:

If the object store were migrate then we would not like to migrate it twice. 

#### Which issue(s) this PR fixes:

Fixes # [sc-70239]

#### Special notes for your reviewer:
Blocked by; https://github.com/replicatedhq/kURL/pull/4239

#### Does this PR introduce a user-facing change?

```release-note
Adds check to not allow perform more than once successfully the object store migration 
```

### CODE REVIEW

File: addons/minio/template/base/install.sh [hunk 1]
```diff
@@ -239,6 +239,12 @@ function minio_migrate_from_rgw() {
...
     minio_wait_for_health
 
+    DID_MIGRATE_ROOK_OBJECT_STORE=$(kubectl -n kurl get configmap kurl-migration-from-rook -o jsonpath='{ .data.DID_MIGRATE_ROOK_OBJECT_STORE }')
+    if [ "$DID_MIGRATE_ROOK_OBJECT_STORE" == "1" ]; then
+        logWarn "Object store is set as migrated previously. Not migrating object store again."
+        return
+    fi
+    
     migrate_rgw_to_minio
     export DID_MIGRATE_ROOK_OBJECT_STORE="1"
```
Comment (emosbaugh) [inline addons/minio/template/base/install.sh:249]:
```suggestion
    DID_MIGRATE_ROOK_OBJECT_STORE=1
```

**Review by emosbaugh — CHANGES_REQUESTED** [review]
small stylistic nit

File: addons/minio/2022-09-25T15-44-53Z/install.sh [hunk 2]
```diff
@@ -190,8 +190,22 @@ function minio_migrate_from_rgw() {
 
     minio_wait_for_health
 
+    export DID_MIGRATE_ROOK_OBJECT_STORE=0
+    DID_MIGRATE_ROOK_OBJECT_STORE=$(kubectl -n kurl get configmap kurl-migration-from-rook -o jsonpath='{ .data.DID_MIGRATE_ROOK_OBJECT_STORE }')
+    if [ "$DID_MIGRATE_ROOK_OBJECT_STORE" == "1" ]; then
+        logWarn "Object store is set as migrated previously. Not migrating object store again."
+        return
+    fi
```
Comment (camilamacedo86) [inline addons/minio/2022-09-25T15-44-53Z/install.sh:198]:
```suggestion
```

File: addons/minio/2022-09-25T15-44-53Z/install.sh [hunk 3]
```diff
@@ -190,8 +190,16 @@ function minio_migrate_from_rgw() {
 
     minio_wait_for_health
 
+
```
Comment (camilamacedo86) [inline addons/minio/2022-09-25T15-44-53Z/install.sh:193]:
```suggestion
```

File: addons/minio/2022-09-25T15-44-53Z/install.sh [hunk 4]
```diff
@@ -190,6 +190,20 @@ function minio_migrate_from_rgw() {
...
     minio_wait_for_health
 
+    export DID_MIGRATE_ROOK_OBJECT_STORE=0
+    DID_MIGRATE_ROOK_OBJECT_STORE=$(kubectl -n kurl get configmap kurl-migration-from-rook -o jsonpath='{ .data.DID_MIGRATE_ROOK_OBJECT_STORE }')
+    if [ "$DID_MIGRATE_ROOK_OBJECT_STORE" == "1" ]; then
+        logWarn "Object store is set as migrated previously. Not migrating object store again."
+        return
+    fi
+
+    export DID_MIGRATE_ROOK_OBJECT_STORE=0
```
Comment (emosbaugh) [inline addons/minio/2022-09-25T15-44-53Z/install.sh:200]: i think you have the code duplicated here

**Review by emosbaugh — CHANGES_REQUESTED** [review]

File: addons/minio/2022-09-25T15-44-53Z/install.sh [hunk 5]
```diff
@@ -190,6 +190,20 @@ function minio_migrate_from_rgw() {
 
     minio_wait_for_health
 
+    export DID_MIGRATE_ROOK_OBJECT_STORE=0
+    DID_MIGRATE_ROOK_OBJECT_STORE=$(kubectl -n kurl get configmap kurl-migration-from-rook -o jsonpath='{ .data.DID_MIGRATE_ROOK_OBJECT_STORE }')
+    if [ "$DID_MIGRATE_ROOK_OBJECT_STORE" == "1" ]; then
+        logWarn "Object store is set as migrated previously. Not migrating object store again."
+        return
+    fi
```
Comment (camilamacedo86) [inline addons/minio/2022-09-25T15-44-53Z/install.sh:198]:
```suggestion
```

File: addons/minio/2022-09-25T15-44-53Z/install.sh (hunk 4 above)
Comment (camilamacedo86) [reply addons/minio/2022-09-25T15-44-53Z/install.sh:194]: I removed that and I think I  pushed again.
 Thank for getting this one.

**Review by emosbaugh — APPROVED** [review]

**Pull request merged**